### ISSUE TITLE
Crash report 7.3: NPE in ReaderSearchService

### ISSUE THREAD

**Issue by maxme** [wordpress-mobile/WordPress-Android#5867]
```
Fatal Exception: java.lang.RuntimeException: Unable to start service org.wordpress.android.ui.reader.services.ReaderSearchService@413f65f0 with Intent { cmp=org.wordpress.android/.ui.reader.services.ReaderSearchService }: java.lang.NullPointerException
       at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:2608)
       at android.app.ActivityThread.access$1900(ActivityThread.java:130)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1308)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:137)
       at android.app.ActivityThread.main(ActivityThread.java:4847)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:535)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
       at dalvik.system.NativeStart.main(NativeStart.java)
Caused by java.lang.NullPointerException
       at libcore.net.UriCodec.encode(UriCodec.java:132)
       at java.net.URLEncoder.encode(URLEncoder.java:57)
       at org.wordpress.android.util.UrlUtils.urlEncode(UrlUtils.java:23)
       at org.wordpress.android.ui.reader.services.ReaderSearchService.startSearch(ReaderSearchService.java:81)
       at org.wordpress.android.ui.reader.services.ReaderSearchService.onStartCommand(ReaderSearchService.java:74)
       at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:2591)
       at android.app.ActivityThread.access$1900(ActivityThread.java:130)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1308)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:137)
       at android.app.ActivityThread.main(ActivityThread.java:4847)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:535)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

**Comment by maxme** [comment]
I'm not sure how this can happen, the query string is null checked and trimmed right before the service is started: https://github.com/wordpress-mobile/WordPress-Android/blob/f32734109acfb40e666889b997f36e560f058cf5/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java#L640-L650

**Comment by maxme** [comment]
The other call to the update method is [here](https://github.com/wordpress-mobile/WordPress-Android/blob/f32734109acfb40e666889b997f36e560f058cf5/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java#L995), it will be called for loading more posts. User can't load more posts, if the search query was null (and the post list empty) 🤔 - I'll null check the query in `ReaderSearchService.onStartCommand`

**Comment by daniloercoli** [comment]
Fixed in #5870
